v2/astradb-to-bigquery/src/test/java/com/google/cloud/teleport/v2/astradb/templates/TestAstraDbToBigQueryIT.java

```diff
@@ -0,0 +1,257 @@
+import static com.google.cloud.teleport.it.common.matchers.TemplateAsserts.assertThatResult;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.Session;
+import com.dtsx.astra.sdk.db.AstraDbClient;
+import com.dtsx.astra.sdk.db.DatabaseClient;
+import com.dtsx.astra.sdk.db.domain.Database;
+import com.dtsx.astra.sdk.db.domain.DatabaseCreationRequest;
+import com.dtsx.astra.sdk.db.domain.DatabaseStatusType;
+import com.dtsx.astra.sdk.utils.ApiLocator;
+import com.google.cloud.bigquery.TableId;
+import com.google.cloud.teleport.it.common.PipelineLauncher;
+import com.google.cloud.teleport.it.common.PipelineOperator;
+import com.google.cloud.teleport.it.common.utils.ResourceManagerUtils;
+import com.google.cloud.teleport.it.gcp.TemplateTestBase;
+import com.google.cloud.teleport.it.gcp.bigquery.BigQueryResourceManager;
+import com.google.cloud.teleport.it.gcp.bigquery.conditions.BigQueryRowsCheck;
+import com.google.cloud.teleport.metadata.TemplateIntegrationTest;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.Serializable;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Integration test for {@link com.google.cloud.teleport.v2.astradb.templates.AstraDbToBigQuery}
+ * (AstraDB_to_BigQuery).
+ */
+@RunWith(JUnit4.class)
+@Category(TemplateIntegrationTest.class)
+@TemplateIntegrationTest(AstraDbToBigQuery.class)
+public class TestAstraDbToBigQueryIT extends TemplateTestBase implements Serializable {
+
+  /** Logger for the class. */
+  private static final Logger LOGGER = LoggerFactory.getLogger(TestAstraDbToBigQueryIT.class);
+
+  /** Pipeline used for this test. */
+  @Rule public transient TestPipeline pipeline = TestPipeline.create();
+
+  /** Default Number of Rows to be inserted in the AstraDB table. */
+  private static final long NUM_ROWS = 500L;
+
+  /*** Astra Region to create a DB (free Tiers). */
+  private static final String TEST_ASTRA_REGION = "us-east1";
+
+  /** Parameters to build the source. */
+  private String astraToken;
+
+  private String astraSecureConnectBundle;
+  private String astraDatabase;
+  private String astraKeyspace;
+  private String astraTable;
+
+  /** Decoding Google Secrets. */
+  private String astraTokenValue;
+
+  private byte[] astraSecureConnectBundleValue;
+
+  private BigQueryResourceManager bigQueryClient;
+
+  @Before
+  public void setup() throws Exception {
+    // Read command line Arguments
+    LOGGER.debug("Initializing Integration Test..");
+    astraToken = System.getProperty("astraToken");
```

Comment:
We'll probably have to ignore the test if those parameters are not given.

We try to use an approach that tests are hermetic / self-contained, which is probably a better fit for contributions. Can we somehow wrap Astradb endpoint as a docker image? See KafkaResourceManager, for example

Reply:
Unfortunately, AFAIK having a docker endpoint is not possible no, Astra is only a Saas.
An alternative could be to have a DB always ON with a read-only token provided as default for this test.

Reply:
I think this is reasonable, but we don't want to cause problems for the average user that makes changes and runs `mvn test / verify`. We'll probably need to ignore if the parameters are not set

like

```
assumeTrue(System.getProperty("astraDatabase") != null);
```